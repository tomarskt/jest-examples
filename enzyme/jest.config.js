module.exports = {
    "verbose": false,
    "bail": true,
    "testEnvironment": "jsdom",
    "setupFiles": ["<rootDir>/setup/shim.js", "<rootDir>/setup/index.js"],
    "snapshotSerializers": ["enzyme-to-json/serializer"]
}