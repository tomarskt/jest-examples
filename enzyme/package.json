{
  "scripts": {
    "test": "jest",
    "debug": "node --inspect-brk node_modules/.bin/jest  --runInBand  "
  },
  "dependencies": {
    "react": "^16.4.0",
    "react-dom": "^16.4.0",
    "prop-types": "^15.6.2"
  },
  "devDependencies": {
    "babel-plugin-transform-async-to-generator": "^6.24.1",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-polyfill": "^6.26.0",
    "babel-preset-es2015": "*",
    "babel-preset-react": "*",
    "enzyme": "^3.3.0",
    "enzyme-adapter-react-16": "^1.0.1",
    "enzyme-to-json": "^3.3.4",
    "jest": "^23.4.1",
    "react-router-dom": "^4.3.1",
    "react-test-renderer": "^16.4.0",
    "redux-form": "^7.3.0",
    "lerna-test-library": "^1.2.1"
  }
}
