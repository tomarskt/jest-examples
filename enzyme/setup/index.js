import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// to supress all warnings, uncomment below line;
// no need to install jsdom or setup jsom, as it's part of jest and 
// jest default to jsdom in testEnvironmenet settins. see package.json jest section
// process.env.NODE_ENV = 'production' 


Enzyme.configure({ adapter: new Adapter() });

