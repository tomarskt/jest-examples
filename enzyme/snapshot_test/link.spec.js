import React from 'react'
import renderer from 'react-test-renderer'
import Link from './Link'
import { shallow, mount } from 'enzyme'

//jest.mock('./Icon', ()=>'Icon') if you have this line, then you will have <Icon tpe='facebook' /> instead of <div><img src='../img.png'/></div> . i.e. Icon is mocked
// no need to call jest.genMockFromModule('fs') either ; 
// no need to create a 'module_name.js' under __mocks__ folder

import IconMock from './Icon'
describe('Link', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <Link href="http://facebook.com/" icon="facebook">
        Facebook
      </Link>
    )

    expect(component.toJSON()).toMatchSnapshot()
    console.log(IconMock)


  })
  it('mount', () => {
    const wrapper = mount(<Link href= "http://facebook.com/" icon= "facebook" >
      Facebook
      </Link >
    )
    console.log(wrapper.html())
})
})