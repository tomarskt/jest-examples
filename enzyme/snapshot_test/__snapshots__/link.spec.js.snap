exports[`Link should render correctly 1`] = `
<a>
  <div>
    <img
      src="../img.png" />
  </div>
  Facebook
</a>
`;
